blocks/main_loop.py

```diff
@@ -113,6 +127,20 @@ def run(self):
         a `training_finish_requested` record in the log.
 
         """
+        # This should do nothing if the user has already configured
+        # logging, and will it least enable error messages otherwise.
+        logging.basicConfig()
+
+        if self._model and isinstance(self.algorithm,
+                                      DifferentiableCostMinimizer):
+            # Sanity check: model and algorithm should be configured
+            # similarly.
+            if not self._model.get_objective() == self.algorithm.cost:
+                raise ValueError("different costs for model and algorithm")
+            if not (set(self._model.get_params().values()) ==
+                    set(self.algorithm.params)):
+                raise ValueError("different params for model and algorithm")
```

Comment:
Sure, it must be warning! My fingers are living a separate live from my brain.. :)